app/src/main/java/com/teamagam/gimelgimel/app/settings/dialogs/SetUsernameAlertDialogBuilder.java

```diff
@@ -48,9 +48,13 @@ public void setOnFinishCallback(Action0 onFinishCallback) {
     private EditText createInputEditText() {
         final EditText input = new EditText(mContext);
         input.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_FLAG_CAP_WORDS);
+        InputFilter[] lengthFilter = new InputFilter[1];
+        lengthFilter[0] = new InputFilter.LengthFilter(R.integer.max_user_name_length);
         input.setFilters(new InputFilter[]{EMOJIS_FILTER});
+        input.setFilters(lengthFilter);
```

Comment:
You should setFilters only once, therefore the use of array. The way you did it you override the emoji filter...
Also extract a method "getFilters()"